watcher/src/watcher_db.rs

```diff
@@ -672,70 +674,66 @@ impl WatcherDB {
-            if block_signer == verification_report_block_signer {
+        // Write no attestation evidence for all the other block signers we missed.
+        for potential_signer in potential_block_signers.iter() {
+            // The `block_signer` gets written together with the
+            // attestation evidence outside of this loop.
+            if potential_signer == block_signer {
                 continue;
             }
 
-            self.write_verification_report(&mut db_txn, src_url, block_signer, None)?;
+            self.write_attestation_evidence(&mut db_txn, src_url, potential_signer, None)?;
         }
 
         // Remove all the keys we encountered from the queue - we no longer need to poll
         // for them.
-        self.remove_verification_report_poll_from_queue(
-            &mut db_txn,
-            src_url,
-            verification_report_block_signer,
-        )?;
+        self.remove_attestation_evidence_poll_from_queue(&mut db_txn, src_url, block_signer)?;
         for block_signer in potential_block_signers.iter() {
-            self.remove_verification_report_poll_from_queue(&mut db_txn, src_url, block_signer)?;
+            self.remove_attestation_evidence_poll_from_queue(&mut db_txn, src_url, block_signer)?;
         }
 
         // Done
         db_txn.commit()?;
         Ok(())
     }
 
-    /// A helper for writing a single (src_url, signer) -> VerificationReport
+    /// A helper for writing a single (src_url, signer) -> attestation_evidence
     /// entry in the database.
-    fn write_verification_report(
+    fn write_attestation_evidence(
         &self,
         db_txn: &mut RwTransaction<'_>,
         src_url: &Url,
         signer: &Ed25519Public,
-        verification_report: Option<&VerificationReport>,
+        attestation_evidence: Option<&VerificationReport>,
     ) -> Result<(), WatcherDBError> {
         let mut key_bytes = signer.to_bytes().to_vec();
         key_bytes.extend(src_url.as_str().as_bytes());
 
-        let value_bytes = verification_report
+        let value_bytes = attestation_evidence
             .map(mc_util_serial::encode)
             .unwrap_or_else(Vec::new);
 
         log::trace!(
             self.logger,
-            "write_verification_report: src_url:{} signer:{} report-provided:{} report-len:{}",
+            "write_attestation_evidence: src_url:{} signer:{} evidence-provided:{} report-len:{}",
             src_url,
             hex::encode(signer.to_bytes()),
-            verification_report.is_some(),
+            attestation_evidence.is_some(),
             value_bytes.len(),
         );
 
-        // First, write the hash -> verification report entry.
-        let hash: [u8; 32] = value_bytes.digest32::<MerlinTranscript>(b"verification_report");
+        // First, write the hash -> attestation evidence entry.
+        let hash: [u8; 32] = value_bytes.digest32::<MerlinTranscript>(b"attestation_evidence");
```

Comment:
My understanding is that updating this value to the hash will only affect if we restart Watcher DB without restarting consensus. This creates a possibility we have 2 entries after a restart. One for `"verification_report"` and one for `"attestation_evidence"`.  It appears we have logic to handle duplicate entries.

The hash only seems to be used to see if the report is already in the DB, so don't think this will affect historical look ups.

Reply:
The hash is also used here (but is not being calculated there): https://github.com/mobilecoinfoundation/mobilecoin/blob/944ababc13831fd57cb91dce0ba959733610b3f6/watcher/src/watcher_db.rs#L833C45-L833C73

but I don't think this will get affected by the hashing being changed, so we're good.